src/detect-quic-cyu-hash.c

```diff
@@ -0,0 +1,428 @@
+        }
+
+        local_id++;
+    }
+}
+
+static void PrefilterMpmQuicHashFree(void *ptr)
+{
+    SCFree(ptr);
+}
+
+static int PrefilterMpmQuicHashRegister(DetectEngineCtx *de_ctx, SigGroupHead *sgh, MpmCtx *mpm_ctx,
+        const DetectBufferMpmRegistery *mpm_reg, int list_id)
+{
+    PrefilterMpmQuicHash *pectx = SCCalloc(1, sizeof(*pectx));
+    if (pectx == NULL)
+        return -1;
+    pectx->list_id = list_id;
+    pectx->mpm_ctx = mpm_ctx;
+    pectx->transforms = &mpm_reg->transforms;
+
+    return PrefilterAppendTxEngine(de_ctx, sgh, PrefilterTxQuicHash, mpm_reg->app_v2.alproto,
+            mpm_reg->app_v2.tx_min_progress, pectx, PrefilterMpmQuicHashFree, mpm_reg->pname);
+}
+
+static bool DetectQuicHashValidateCallback(const Signature *s, const char **sigerror)
+{
+    const SigMatch *sm = s->init_data->smlists[g_buffer_id];
+    for (; sm != NULL; sm = sm->next) {
+        if (sm->type != DETECT_CONTENT)
+            continue;
+
+        const DetectContentData *cd = (DetectContentData *)sm->ctx;
+
+        if (cd->flags & DETECT_CONTENT_NOCASE) {
+            *sigerror = BUFFER_NAME " should not be used together with "
+                                    "nocase, since the rule is automatically "
+                                    "lowercased anyway which makes nocase redundant.";
+            SCLogWarning(SC_WARN_POOR_RULE, "rule %u: %s", s->id, *sigerror);
+        }
+
+        if (cd->content_len != 32) {
+            *sigerror = "Invalid length of the specified" BUFFER_NAME " (should "
+                        "be 32 characters long). This rule will therefore "
+                        "never match.";
+            SCLogWarning(SC_WARN_POOR_RULE, "rule %u: %s", s->id, *sigerror);
+            return FALSE;
+        }
+        for (size_t i = 0; i < cd->content_len; ++i) {
+            if (!isxdigit(cd->content[i])) {
+                *sigerror = "Invalid " BUFFER_NAME
+                            " string (should be string of hexadecimal characters)."
+                            "This rule will therefore never match.";
+                SCLogWarning(SC_WARN_POOR_RULE, "rule %u: %s", s->id, *sigerror);
+                return FALSE;
+            }
+        }
+    }
+
+    return TRUE;
+}
+
+void DetectQuicCyuHashRegister(void)
+{
+    /* quic.cyu.hash sticky buffer */
+    sigmatch_table[DETECT_AL_QUIC_CYU_HASH].name = KEYWORD_NAME;
+    sigmatch_table[DETECT_AL_QUIC_CYU_HASH].desc = "sticky buffer to match on the QUIC CYU hash";
+    sigmatch_table[DETECT_AL_QUIC_CYU_HASH].url = "/rules/" KEYWORD_DOC;
+    sigmatch_table[DETECT_AL_QUIC_CYU_HASH].Setup = DetectQuicCyuHashSetup;
+    sigmatch_table[DETECT_AL_QUIC_CYU_HASH].flags |= SIGMATCH_NOOPT;
+#ifdef UNITTESTS
+    sigmatch_table[DETECT_AL_QUIC_CYU_HASH].RegisterTests = DetectQuicCyuHashRegisterTests;
+#endif
+
+    DetectAppLayerMpmRegister2(
+            BUFFER_NAME, SIG_FLAG_TOSERVER, 2, PrefilterMpmQuicHashRegister, NULL, ALPROTO_QUIC, 1);
```

Comment:
I think you can use generic `PrefilterGenericMpmRegister` cf https://github.com/OISF/suricata/pull/6270#discussion_r700104353

Reply:
we don't have a generic one for multi-buffers yet